test/sanity/code-smell/no-illegal-filenames.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+
+# a script to check for illegal filenames on various Operating Systems. The
+# main rules are derived from restrictions on Windows
+# https://msdn.microsoft.com/en-us/library/aa365247#naming_conventions
+
+import os
+import struct
+
+illegal_chars = [
+    b'<',
+    b'>',
+    b':',
+    b'"',
+    b'/',
+    b'\\',
+    b'|',
+    b'?',
+    b'*'
+]
```

Comment:
This could be simplified by by using the following when constructing `illegal_chars`:

```python
] + [struct.pack("b", i) for i in range(32)]
```